cpp/include/cudf/scalar/scalar.hpp

```diff
@@ -73,7 +73,7 @@ class scalar {
    * @param is_valid true: set the value to valid. false: set it to null
    * @param stream CUDA stream used for device memory operations.
    */
-  void set_valid(bool is_valid, rmm::cuda_stream_view stream = rmm::cuda_stream_default);
+  void set_valid_async(bool is_valid, rmm::cuda_stream_view stream = rmm::cuda_stream_default);
```

Comment:
I think the r-value version of this function needs to be deleted?

Reply:
Actually I think it should not be deleted. Doing so results in ambiguity between the `bool` and `bool&&` versions of the function. The reason to delete would be to avoid the case where a reference to a temporary is passed to an async function, and the temporary is destroyed before the copy completes. But `bool` is a special case for which the lowest-level implementation in `device_uvector` uses `cudaMemset`. And since both `set_valid_async` and `cudaMemset` both take non-reference parameters, the value of the temporary will be copied into `cudaMemsetAsync` before the temporary can be destroyed.